audioipc/src/lib.rs

```diff
@@ -168,10 +168,15 @@ unsafe fn close_platformhandle(handle: PlatformHandleType) {
     winapi::um::handleapi::CloseHandle(handle);
 }
 
-pub fn get_shm_path(dir: &str) -> PathBuf {
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+static SHM_ID: AtomicUsize = AtomicUsize::new(0);
+
+pub fn get_shm_path() -> PathBuf {
     let pid = std::process::id();
+    let shm_id = SHM_ID.fetch_add(1, Ordering::SeqCst);
     let mut temp = temp_dir();
-    temp.push(&format!("cubeb-shm-{}-{}", pid, dir));
+    temp.push(&format!("cubeb-shm-{}-{}", pid, shm_id));
```

Comment:
We were depending on Unix-like temporary file delete behaviour where the file immediately disappears from the filesystem, but on Windows/NTFS the file can be queued for deletion but still visible in some circumstances.  In that case, we could end up attempting to reuse a file when we intended to create a new temporary file - when that happened, we'd hit an error because we open our temporary files with `create_new`, which fails if a file already exists with the specified name.

Reply:
I'll add a comment to get_shm_path.